--- cgic.h	2004-04-22 16:49:47.000000000 -0400
+++ ../jsonptunnel/cgic/cgic.h	2008-12-16 15:35:13.000000000 -0500
@@ -9,7 +9,7 @@
 /* Bring in standard I/O since some of the functions refer to
 	types defined by it, such as FILE *. */
 
-#include <stdio.h>
+#include <fcgi_stdio.h>
 
 /* The various CGI environment variables. Instead of using getenv(),
 	the programmer should refer to these, which are always
