--- cgic.c	2004-11-15 08:57:59.000000000 -0800
+++ cgic.c.patched	2008-12-18 08:19:02.000000000 -0800
@@ -120,7 +120,11 @@
 static int cgiStrBeginsNc(char *s1, char *s2);
 
 int main(int argc, char *argv[]) {
+  #ifdef USE_FASTCGI 
+  while(FCGI_Accept() >= 0) {
+  #else
 	int result;
+  #endif
 	char *cgiContentLengthString;
 	char *e;
 	cgiSetupConstants();
@@ -284,9 +288,17 @@
 #endif /* CGICDEBUG */
 		}
 	}
+  #ifdef USE_FASTCGI 
+  cgiMain();
+  cgiFreeResources();
+  } 
+
+  return 1;
+  #else
 	result = cgiMain();
 	cgiFreeResources();
 	return result;
+  #endif
 }
 
 static void cgiGetenv(char **s, char *var){
