--- /Users/alan/Projects/src/cgic205/cgic.c	2004-11-15 08:57:59.000000000 -0800
+++ cgic/cgic.c	2008-12-18 11:13:58.000000000 -0800
@@ -22,7 +22,11 @@
 #define CGICDEBUGEND
 #endif /* CGICDEBUG */
 
+#ifdef USE_FASTCGI 
+#include <fcgi_stdio.h>
+#else
 #include <stdio.h>
+#endif
 #include <string.h>
 #include <ctype.h>
 #include <stdlib.h>
@@ -120,7 +124,11 @@
 static int cgiStrBeginsNc(char *s1, char *s2);
 
 int main(int argc, char *argv[]) {
+  #ifdef USE_FASTCGI 
+  while(FCGI_Accept() >= 0) {
+  #else
 	int result;
+  #endif
 	char *cgiContentLengthString;
 	char *e;
 	cgiSetupConstants();
@@ -284,9 +292,17 @@
 #endif /* CGICDEBUG */
 		}
 	}
+  #ifdef USE_FASTCGI 
+  cgiMain();
+  cgiFreeResources();
+  } 
+
+  return 1;
+  #else
 	result = cgiMain();
 	cgiFreeResources();
 	return result;
+  #endif
 }
 
 static void cgiGetenv(char **s, char *var){
